docs: dashboard plugins SDK auto upgrades (#4333)
gooddata/gooddata-ui-sdk: docs/content/en/latest/references/dashboard_component/dashboard_plugins.md

### Recommendations
- Use only public dashboard plugins API and GoodData.UI interfaces. Using anything marked as alpha, beta or internal in the plugin code is at your own risk. If you are missing API for your particular use-case, or something you need is not marked as a stable yet, please [get in touch with us](../../../#Home-JoinLearn) and we may consider to update or extend it.
- Do not manipulate any CSS or DOM structure of the dashboard, rather stick with the [theming](https://www.gooddata.com/docs/cloud/customize-appearance/create-custom-themes/).
- Create smaller focused plugins, avoid large complicated plugins that are drastically changing the dashboard behavior and look.
- For more complex changes, consider better suited approaches: [using standalone dashboard component](../../../learn/embed_dashboards/#EmbedDashboards-Dashboardcomponent) or [building your own custom dashboard](../../../learn/embed_dashboards/#EmbedDashboards-Customdashboards).

### Maintenance

> As the dashboard plugins are mostly integrated with our SaaS platform (GoodData Platform or GoodData Cloud) that is ever evolving, sometimes there is a need to introduce breaking-changes. 

Reviewer: ```suggestion
Since dashboard plugins are primarily integrated with our ever-evolving SaaS platform (GoodData Platform or GoodData Cloud), there are occasions when introducing breaking changes is necessary.
```

> When this happens, there will be a graceful period ~6 months that will allow you to safely migrate your plugins, however after that time period, dashboard starts to run with the latest version by default and it won’t be possible to switch back.

Reviewer: ```suggestion
When this occurs, a grace period of approximately six months will be provided, allowing for the safe migration of your plugins. However, after this time frame, the dashboard will default to running with the latest version, and reverting to older versions will not be possible. Any plugins that have not been migrated to the new API version will not load once the grace period concludes.
```

> Plugins that were not migrated to a newer API version won’t be loaded after the graceful period ends.

Reviewer: ```suggestion
```
